pkgs/development/libraries/glib-networking/default.nix

```diff
@@ -1,49 +1,93 @@
-{ stdenv, fetchurl, meson, ninja, pkgconfig, glib, gettext, python3, gnutls, p11-kit, libproxy, gnome3
-, gsettings-desktop-schemas }:
+{ stdenv
+, fetchurl
+, substituteAll
+, meson
+, ninja
+, nixosTests
+, pkgconfig
+, glib
+, gettext
+, makeWrapper
+, python3
+, gnutls
+, p11-kit
+, libproxy
+, gnome3
+, gsettings-desktop-schemas
+}:
 
-let
+stdenv.mkDerivation rec {
   pname = "glib-networking";
   version = "2.60.3";
-in
-stdenv.mkDerivation rec {
-  name = "${pname}-${version}";
+
+  outputs = [ "out" "installedTests" ];
 
   src = fetchurl {
-    url = "mirror://gnome/sources/${pname}/${stdenv.lib.versions.majorMinor version}/${name}.tar.xz";
+    url = "mirror://gnome/sources/${pname}/${stdenv.lib.versions.majorMinor version}/${pname}-${version}.tar.xz";
     sha256 = "1mfw44qpmwvz6yzj8c6spx6z357wrmkk15byrkc5byagd82860fm";
   };
 
-  outputs = [ "out" "dev" ]; # to deal with propagatedBuildInputs
+  patches = [
+    (substituteAll {
+      src = ./hardcode-gsettings.patch;
+      gds_gsettings_path = glib.getSchemaPath gsettings-desktop-schemas;
+    })
 
-  PKG_CONFIG_GIO_2_0_GIOMODULEDIR = "${placeholder "out"}/lib/gio/modules";
+    ./installed-tests-path.patch
+  ];
 
   postPatch = ''
     chmod +x meson_post_install.py # patchShebangs requires executable file
     patchShebangs meson_post_install.py
   '';
 
   nativeBuildInputs = [
-    meson ninja pkgconfig gettext
-    python3 # install_script
+    meson
+    ninja
+    pkgconfig
+    gettext
+    makeWrapper
+    python3 # for install_script
   ];
-  propagatedBuildInputs = [ glib gnutls p11-kit libproxy gsettings-desktop-schemas ];
 
-  mesonFlags = [
-    # Default auto detection doesn't work
-    "-Dgnutls=enabled"
+  buildInputs = [
+    glib
+    gnutls
+    p11-kit
+    libproxy
+    gsettings-desktop-schemas
   ];
 
   doCheck = false; # tests need to access the certificates (among other things)
 
+  mesonFlags = [
+    "-Dinstalled_tests=true"
+    "-Dinstalled_test_prefix=${placeholder "installedTests"}"
+  ];
+
+  postFixup = ''
+    find "$installedTests/libexec" -type f -executable -print0 \
+      | while IFS= read -r -d "" file; do
+      echo "Wrapping program '$file'"
+      wrapProgram "$file" --prefix GIO_EXTRA_MODULES : "$out/lib/gio/modules"
```

Comment:
Why just the gio modules? Couldn't we use `wrapGAppsHook`, disable it, and wrap them with its args?

Reply:
I do not want to introduce dependencies on GTK, librsvg etc.

Reply:
I do not want to introduce dependencies on GTK, librsvg etc.

Reply:
Right, can we make note of that so someone doesn't refactor it to this down the line?

Reply:
Can we fix the assumptions about `wrapGAppsHook` instead? I really need to finish https://github.com/NixOS/nixpkgs/pull/43150

Reply:
> I really need to finish #43150

I do have a notes locally on some things I wanted to add to https://github.com/NixOS/nixpkgs/pull/43150 but documentation writing escapes me also.
Though it would be good, just even as a start, to merge the parts about the `wrapGAppsHook` which I believe was your original intent with this.

> Can we fix the assumptions about wrapGAppsHook instead? It even has apps in the name.

When you say "it has apps in the name", it's confusing when it would be used for executables in a library?

Reply:
> I do have notes locally on some things I wanted to add to #43150 but documentation writing escapes me also.

I guess I will try to work on I today.

> Though it would be good, just even as a start, to merge the parts about wrapGAppsHook which I believe was your original intent with this. 

Yeah keeping the scope narrow will make it easier to consider it mergeable. It is just bad that the wrapper connects so much stuff.

> When you say "it has apps in the name", it's confusing when it would be used for executables in a library?

For me, an app is primarily a graphical program that a person can use to carry out some task. Though there can be console apps for which we would not want to depend on GTK either. I would not consider such helper of library to be an app.

Reply:
> > I do have notes locally on some things I wanted to add to #43150 but documentation writing escapes me also.
> 
> I guess I will try to work on I today.
> 
> > Though it would be good, just even as a start, to merge the parts about wrapGAppsHook which I believe was your original intent with this.
> 
> Yeah keeping the scope narrow will make it easier to consider it mergeable. It is just bad that the wrapper connects so much stuff.

Great, I'd be happy to follow up on it once it's merged. And our roadmap can include what needs to be documented in the future.